### PR TITLE
Use log4j version 2

### PR DESCRIPTION
**Pull request by supl**
This PR upgrades log4j to version 2 for vulnerability fix.

### CODE REVIEW

File: java/build.gradle [hunk 1]
```diff
@@ -16,9 +16,13 @@ repositories {
...
 def grpcVersion = '1.43.2'
 def protobufVersion = '3.19.2'
+def slf4jVersion = '1.7.36'
+def log4jVersion = '2.19.0'
 
 dependencies {
-    compile group: 'org.slf4j', name: 'slf4j-log4j12', version: '1.7.30'
+    compile group: 'org.slf4j', name: 'slf4j-api', version: "${slf4jVersion}"
+    compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: "${log4jVersion}"
+    compile group: 'org.apache.logging.log4j', name: 'log4j-slf4j-impl', version: "${log4jVersion}"
```
Comment (brfrn169) [inline java/build.gradle:25]: I think we should use `log4j-slf4j2-impl` for SLF4j 2.0.
```suggestion
    compile group: 'org.apache.logging.log4j', name: 'log4j-slf4j2-impl', version: "${log4jVersion}"
```

**Review by brfrn169 — COMMENTED** [review]
Left a comment. Please take a look when you have time!

File: java/build.gradle (hunk 1 above)
Comment (brfrn169) [reply java/build.gradle:25]: Ah, sorry. I misunderstood this PR upgrades slf4j to 2.0. Please ignore it.

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**